topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -1229,6 +1249,10 @@ public void setKafkaTopics(List<KafkaTopic> ktList) {
         }).compose(reconcileState -> {
             List<Future> futs = new ArrayList<>();
             topicCounter.set(reconcileState.ktList.size());
+            if (reconcileState.ktList.size() == 0) {
+                reconciliationsCounter.increment();
+                successfulReconciliationsCounter.increment();
+            }
             for (KafkaTopic kt : reconcileState.ktList) {
```

Comment:
Why does this get counted as a (successful) reconciliation, but when there are topics each one counts as successful? What are we counting? The number of times the timer goes off or the number of time we actually deal with a topic?

Reply:
Ah, it should not be there.